assignment4/assignment4.cpp

```diff
@@ -0,0 +1,54 @@
+#include <iostream>
+#include <math.h>
+#include <vector>
+
+class IslandCounter{
+	void collapse_island(std::vector< std::vector<bool> > &map, int i, int j){
+		if(!map[i][j]) return;
+
+		map[i][j] = false;
+		collapse_island( map, std::min(i+1, (int) (map.size()-1)), j);
+		collapse_island( map, std::max(i-1, 0), j);
+		collapse_island( map, i, std::min(j+1,(int) (map[i].size()-1)));
+		collapse_island( map, i, std::max(j-1,0));
+		
+		return;
+	}
+public:
+	int Count(std::vector< std::vector<bool> > map){
+		int island_counter = 0;
+		for( int i=0; i<map.size(); i++) {
```

Comment:
I think the code looks better if every arithmetic operation has spaces before and after itself. Also every operator has a space before {} and (). For example:
`for (int i = 0; i < N; i++) {`
`       a += 5;`
`}`